### ISSUE TITLE
How to: Create an environment variable without a shell command (shared server)

### ISSUE THREAD

**Issue by fsimonin** [sendgrid/sendgrid-php#880]
Hello

I'm struggling at sending my first email with Sendgrid in a Php environment (Api v3).
In the "Integrate using our Web API" guide, you ask to "Create an environment variable" by "running in the shell" a command. But as I use a shared server, I can't access to a shell.

So is there another way to set/create an environment variable?

Thx.

PHP Version: 7.0
Server : Mutualized server on Online.net

**Comment by Virneto** [comment]
...same here.. :(
also, setting an environment variable means I'll have to set it again everytime my server goes down and rebots..
Is there any alternative to setting the APIKEY?
I've tried  this but with no luck..
`$sendgrid = new \SendGrid($myAPIkey);`

**Comment by DaWe35** [comment]
I'm also using
`$sendgrid = new \SendGrid('COPY-PASTED-API-KEY');`
and it's working for me.

**Comment by jhalag** [comment]
An ENV variable is not strictly necessary - although it is a best practice to separate your API keys and such from your code. In situations such as you mentioned (shared hosts) you can skip the getenv() call and either import it from a credentials file (if you are using such), or just paste it right in the code as a string literal (as @DaWe35 posted).

_Friendly reminder: Be careful not to commit any code containing API keys or other credentials to publicly-accessible source control (such as GitHub)_

**Comment by fsimonin** [comment]
Thx for your time and answers @jhalag, @DaWe35, @Virneto.
On my side, it's still not work. I tried to put the api key directly in my php file.

So my following question is about the line "require 'vendor/autoload.php'; // If you're using Composer"
I'm not using vendor, again because of a shared server, so I don't really know how to use or if it's available.
Does it work for you because you're using Vendor ?

In my real example on my website, I don't have any feedback when I trigger to send an email.
With the example.php available in the folder sendgrid, I have at least a message saying "Authorization required"

Thx for any idea/help

**Comment by DaWe35** [comment]
>  Does it work for you because you're using Vendor ?

I'm not using Composer. I recommend you to download the latest release (https://github.com/sendgrid/sendgrid-php/releases/download/v7.3.0/sendgrid-php.zip), and rebuild the code from the tutorial.

**Comment by Robin-the-Frog** [comment]
Don't know if this helps, but is it to do with this section at the top of the code...

```
require 'vendor/autoload.php'; // If you're using Composer (recommended)
// Comment out the above line if not using Composer
// require("<PATH TO>/sendgrid-php.php");
// If not using Composer, uncomment the above line and
// download sendgrid-php.zip from the latest release here,
// replacing <PATH TO> with the path to the sendgrid-php.php file,
// which is included in the download:
// https://github.com/sendgrid/sendgrid-php/releases
```

I'm not using Composer, so mine looks like this

```
// require 'vendor/autoload.php'; // If you're using Composer (recommended)
// Comment out the above line if not using Composer
require("sendgrid/sendgrid-php.php");
// If not using Composer, uncomment the above line and
// download sendgrid-php.zip from the latest release here,
// replacing <PATH TO> with the path to the sendgrid-php.php file,
// which is included in the download:
// https://github.com/sendgrid/sendgrid-php/releases
```

And it's `require("sendgrid/sendgrid-php.php");` because I've saved the entire sendgrid directory into
.../public_html/sendgrid on my web server

Then I use a $variable in
`$sendgrid = new \SendGrid($api_key);`
just so that I can set the key elsewhere in my code

Does that help?

**Comment by thinkingserious** [comment]
Hello @fsimonin, @Virneto,

Please let us know if you are still having trouble. 

Thanks to @DaWe35, @jhalag and @Robin-the-Frog for helping out!!!

With best regards,

Elmer

**Comment by fsimonin** [comment]
Hi @thinkingserious and thx to everyone that helped.

I finaly got back on this issue and I was able to resolve it.
I restarted from scratch with an empty page, just at using the template and set the api_key directly in my file for now.

After a few succeeded emails sent, the hotmail server blocked my request, I'll dig around if I can find some information, otherwise I'll open another thread to discuss about it.

Thx
Fred
